src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -154,8 +159,28 @@ public Storage storageForBucket(GoogleBucketResource bucketResource) {
   public Stream<String> getBlobsLinesStream(
       String blobUrl, String cloudEncapsulationId, AuthenticatedUserRequest userRequest) {
     Storage storage = gcsProjectFactory.getStorage(cloudEncapsulationId);
-    return listGcsFiles(blobUrl, cloudEncapsulationId, storage)
-        .flatMap(blob -> getBlobLinesStream(blob, cloudEncapsulationId, storage));
+    try {
+      return listGcsFiles(blobUrl, cloudEncapsulationId, storage)
+          .flatMap(blob -> getBlobLinesStream(blob, cloudEncapsulationId, storage));
+    } catch (StorageException e) {
+      if (e.getCode() == HttpStatus.SC_FORBIDDEN) {
+        String bucket = GcsUriUtils.parseBlobUri(blobUrl).getBucket();
```

Comment:
why?  the idea is to catch a subset of the storageexception (since Google uses broad exception types).  The throw here is just to let the exception pass through

Reply:
The thought experiment here is, what if `GcsUriUtils.parseBlobUri()` throws? If it did, then the user would never see the underlying `StorageException`. We can claim that this will never occur, since we know that the blob URL has already been parsed before this `catch` clause runs, but the code itself doesn't assert that. Calling `parseBlobUrl()` before the `try` and having all downstream code use the parsed blob, makes an implicit property of the code explicit.